### PR TITLE
[Ready-to-Merge] Add great-expecations to conda-forge

### PR DESCRIPTION
**Pull request by sodre**
Great Expectations is a framework that helps teams save time and promote analytic integrity with a new twist on automated testing: pipeline tests. Pipeline tests are applied to data (instead of code) and at batch time (instead of compile or deploy time).

https://github.com/great-expectations/great_expectations

### CODE REVIEW

**Comment by sodre** [comment]
@jpcampb2, would you like to be added as a maintainer of the great-expectations conda-forge package?

File: recipes/great-expectations/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name|replace("-","_") }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  #noarch: python
```
Comment (sodre) [inline recipes/great-expectations/meta.yaml]: can packages that depend on six use noarch?

File: recipes/great-expectations/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{% set name = "great-expectations" %}
```
Comment (rth) [inline recipes/great-expectations/meta.yaml]: The name on PyPi is "great_expectations". I'm not sure if makes sure to keep it for consistency, even if a dash would have been better. For instance `memory_profiler` on PyPi is still `memory_profiler` on conda..

File: recipes/great-expectations/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+build:
+  #noarch: python
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  build:
+    - python
+    - pip
+    - six >=1.9.0
```
Comment (rth) [inline recipes/great-expectations/meta.yaml]: six is not imported in `setup.py` unless I missed it

**Comment by rth** [comment]
Also if possible please add a link to the project github page in the initial PR description (so that reviewers wouldn't have to search for it).

**Comment by jpcampb2** [comment]
Thank you, @sodre, yes please add me as a maintainer!

The github address for the project is https://github.com/great_expectations/great-expectations, as (now) listed in the PR message.

**Comment by sodre** [comment]
@jpcampb2 after the PR is merged you should be able to install with `conda install -c conda-forge great_expectations`. 

Would you rather have a '-' instead of '_'? If so I need to undo one of the reviewer's request.

**Comment by sodre** [comment]
@conda-forge-admin, please rerender

**Comment by jpcampb2** [comment]
@sodre : pip and pypi allow either formulation (great_expectations and great-expectations), so whichever option makes it easier to use is preferred...

File: recipes/great-expectations/meta.yaml (hunk 2 above)
Comment (sodre) [reply recipes/great-expectations/meta.yaml]: @rth, 
I added the upstream developer as a maintainer and it looks like he is fine either way. Feel free to change it when you merge the PR if you still feel strongly about it.

File: recipes/great-expectations/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+test:
+  imports:
+    - great_expectations
+
+about:
+  home: https://github.com/great-expectations/great_expectations
+  license: Apache-2.0
+  license_family: Apache
+  # License is still missing
+  #license_file: LICENSE
```
Comment (rth) [inline recipes/great-expectations/meta.yaml]: It would be probably necessary to [package the LICENSE file manually](https://conda-forge.org/docs/meta.html#packaging-the-license-manually) (it's a requirement for Apache License).

**Comment by rth** [comment]
>> The name on PyPi is "great_expectations". 
>
> I added the upstream developer as a maintainer and it looks like he is fine either way. Feel free to change it when you merge the PR if you still feel strongly about it.

Once this is merged I will be much more difficult to change the name. 

> so whichever option makes it easier to use is preferred...

It's not so much about being easier to use, as about overall naming consistency. I'm not sure what's conda-forge position on that.

I won't be able to merge this because I don't have merge permissions :) Just helping with the reviews...

Generally, LGTM once the missing license file is added. Thanks for the PR @sodre !

File: recipes/great-expectations/meta.yaml (hunk 4 above)
Comment (rth) [reply recipes/great-expectations/meta.yaml]: It would be nice if this could be addressed upstream..

**Comment by rth** [comment]
> The name on PyPi is "great_expectations"

Actually, the  [name on PyPi](https://pypi.org/project/great-expectations) now appears to be with a dash (the .tar.gz name is still with an underscore). I'm not sure if it was changed on PyPi recently (if that's possible) or if I missed something. In any case this would remove the naming issue..

**Comment by jpcampb2** [comment]
The LICENSE file is present on the repo, and I've updated our manifest to push it on the next build.

**Comment by sodre** [comment]
@isuruf, this one has already been reviewed by @rth and should be ready to merge as well.

File: recipes/great-expectations/00-force-long_description.patch [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+index 6877517..18df129 100644
+--- a/setup.py
++++ b/setup.py
+@@ -7,8 +7,7 @@ except ImportError:
+ with open('requirements.txt') as f:
+     required = f.read().splitlines()
+ 
+-with open('docs/source/intro.rst') as f:
+-    long_description = f.read()
++long_description = ''
```
Comment (isuruf) [inline recipes/great-expectations/00-force-long_description.patch]: Why is this needed?

Comment (sodre) [reply recipes/great-expectations/00-force-long_description.patch]: The upstream code is not shipping the "docs" folder so setup.py fails with that line in there.

This is is fixed in their master branch and will be removed in the next release.
